src/components/Wizard/CreateVmWizard/providers/tests/VMWarePasswordAndCheck.test.js

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import { mount, render } from 'enzyme';
+
+import { WithResources } from '../../../../../tests/k8s';
+import VMWarePasswordAndCheck from '../VMWarePasswordAndCheck';
+import fixture from '../fixtures/VMWarePasswordAndCheck.fixture';
+import {
+  PROVIDER_STATUS_CONNECTING,
+  PROVIDER_STATUS_CONNECTION_FAILED,
+  PROVIDER_STATUS_SUCCESS,
+  PROVIDER_VMWARE_CONNECTION,
+  PROVIDER_VMWARE_USER_PWD_AND_CHECK_KEY,
+} from '../../constants';
+import { basicSettingsImportVmwareNewConnection } from '../../../../../tests/forms_mocks/basicSettings.mock';
+
+const rendersProviderStatus = (providerStatus, phase) => {
+  const onChange = jest.fn();
+  const { value } = basicSettingsImportVmwareNewConnection[PROVIDER_VMWARE_USER_PWD_AND_CHECK_KEY];
+  value[PROVIDER_VMWARE_CONNECTION].providerStatus = providerStatus;
+
+  const extraProps = {
+    onCheckConnection: () => {},
+    WithResources: props => <WithResources {...props} tesOnlyPhase={phase} />,
```

Comment:
The WithResources component is a mock. Mocks usually contain hacks to by-pass additional complex, not-tested, logic. IMO, this is the case.

Reply:
yes mocking is fine. I mean it would be easier to have components without WithResources so we wouldn't have to do this. WithResources should be just a slim wrapper which wouldn't need much testing.